lib/src/worker/worker_connection.dart

```diff
@@ -57,6 +65,35 @@ class StdAsyncWorkerConnection implements AsyncWorkerConnection {
   }
 }
 
+/// Implementation of [AsyncWorkerConnection] for running in an isolate.
+class SendPortAsyncWorkerConnection implements AsyncWorkerConnection {
+  final ReceivePort receivePort;
+  final StreamIterator receivePortIterator;
+  final SendPort sendPort;
+
+  factory SendPortAsyncWorkerConnection(SendPort sendPort) {
+    var receivePort = new ReceivePort();
+    sendPort.send(receivePort.sendPort);
+    return SendPortAsyncWorkerConnection._(receivePort, sendPort);
+  }
+
+  SendPortAsyncWorkerConnection._(this.receivePort, this.sendPort)
+      : receivePortIterator = new StreamIterator(receivePort);
+
+  @override
+  Future<WorkRequest> readRequest() async {
+    if (!await receivePortIterator.moveNext()) {
+      throw new StateError('Receive port closed.');
```

Comment:
nit: the other implementation just returns null if the connection closes - that isn't necessarily better but we should remain consistent

Reply:
Works for me--done. Thanks.